learning_observer/learning_observer/auth/utils.py

```
@@ -99,13 +101,28 @@ async def get_active_user(request):
     return session[constants.USER]
 
 
+def google_stored_auth():
+    '''This is a blank function for key making purposes.
+    '''
+    pass
+
+
 async def logout(request):
     '''
     Log the user out
     '''
     session = await aiohttp_session.get_session(request)
-    session.pop(constants.USER, None)
+    user = session.pop(constants.USER, None)
     session.pop(constants.AUTH_HEADERS, None)
+
+    # Remove auth header info from redis
+    kvs = learning_observer.kvs.KVS()
+    remove_auth_key = sa_helpers.make_key(
+        google_stored_auth,
+        {sa_helpers.KeyField.TEACHER: user['user_id']},
+        sa_helpers.KeyStateType.INTERNAL)
+    await kvs.remove(remove_auth_key)
+
```

I would probably remove this when auth expires (e.g. we make a request and it fails with an auth exception), rather than on logout.

---

I believe we will also get an auth exception when the document isn't shared which is likely to happen. I'll see about on adding this to the workflow when the auth expires.

---

I would expect different auth exceptions (even in the text if not the type), but it's worth double-checking. 

We don't want to keep making requests when auth has expired, and we do want to keep track of access permissions.

---

if we get an auth error, stop trying that document.